helper/resource/testing.go

```diff
@@ -481,32 +493,19 @@ func Test(t testing.T, c TestCase) {
 
 	logging.SetOutput()
 
-	// get instances of all providers, so we can use the individual
-	// resources to shim the state during the tests.
-	providers := make(map[string]*schema.Provider)
-	var provider string
-	for name, pf := range c.ProviderFactories {
-		p, err := pf()
-		if err != nil {
-			t.Fatal(err)
-		}
-		providers[name] = p
-		provider = name
-	}
-	for name, p := range c.Providers {
-		providers[name] = p
-		provider = name
-	}
-
-	if len(providers) != 1 {
-		t.Fatalf("Only the provider under test should be set in TestCase, got %d providers. Other providers can be used by adding their provider blocks to their config; they will automatically be downloaded as part of terraform init.", len(providers))
-	}
+	// Copy any explicitly passed providers to factories, this is for backwards compatibility.
+	if len(c.Providers) > 0 {
+		c.ProviderFactories = map[string]func() (*schema.Provider, error){}
 
-	// Auto-configure all providers.
-	for _, p := range providers {
-		diags := p.Configure(context.Background(), terraform.NewResourceConfigRaw(nil))
-		if diags.HasError() {
-			t.Fatal("error configuring provider: %s", diagutils.ErrorDiags(diags))
+		t.Log("TestCase.Providers is deprecated, please use ProviderFactories")
```

Comment:
I would avoid logging the deprecation. I think we can rely on the godoc deprecatiob